zul/src/archive/web/js/zul/db/datefmt.js

```diff
@@ -371,7 +371,7 @@ zk.fmt.Date = {
 					if (!hasHour1)
 						break;
 					if (!token) return; //failed
-					isAM = token.startsWith(localizedSymbols.APM[0]);
+					isAM = token.toUpperCase().startsWith(localizedSymbols.APM[0]);
```

Comment:
Both token and APM[0]`.toUpperCase()` are safer

Reply:
ok